File: samples/samples-java/src/main/java/com/function/AddProductsWithIdentityColumnArray.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.function;

import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.OutputBinding;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;
import com.microsoft.azure.functions.sql.annotation.SQLOutput;
import com.function.Common.ProductWithoutId;

import java.util.Optional;

public class AddProductsWithIdentityColumnArray {
    @FunctionName("AddProductsWithIdentityColumnArray")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = { HttpMethod.GET },
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "addproductswithidentitycolumnarray") HttpRequestMessage<Optional<String>> request,
            @SQLOutput(
                commandText = "dbo.ProductsWithIdentity",
                connectionStringSetting = "SqlConnectionString")
                OutputBinding<ProductWithoutId[]> products) {

        ProductWithoutId[] p = new ProductWithoutId[] {
            new ProductWithoutId("Cup", 2),
            new ProductWithoutId("Glasses", 12)
        };
        products.setValue(p);

        return request.createResponseBuilder(HttpStatus.OK).header("Content-Type", "application/json").body(products).build();
    }
}

File: samples/samples-outofproc/OutputBindingSamples/AddProductsWithIdentityColumnArray.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using Microsoft.Azure.Functions.Worker.Http;
using Microsoft.Azure.WebJobs.Extensions.Sql.SamplesOutOfProc.Common;
using Microsoft.Azure.Functions.Worker.Extensions.Sql;
using Microsoft.Azure.Functions.Worker;

namespace Microsoft.Azure.WebJobs.Extensions.Sql.SamplesOutOfProc.OutputBindingSamples
{
    public static class AddProductsWithIdentityColumnArray
    {
        /// <summary>
        /// This shows an example of a SQL Output binding where the target table has a primary key
        /// which is an identity column. In such a case the primary key is not required to be in
        /// the object used by the binding - it will insert a row with the other values and the
        /// ID will be generated upon insert.
        /// </summary>
        /// <param name="req">The original request that triggered the function</param>
        /// <returns>The new product objects that will be upserted</returns>
        [Function(nameof(AddProductsWithIdentityColumnArray))]
        [SqlOutput("dbo.ProductsWithIdentity", "SqlConnectionString")]
        public static ProductWithoutId[] Run(
            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")]
            HttpRequestData req)
        {
            ProductWithoutId[] products = new[]
            {
                new ProductWithoutId
                {
                    Name = "Cup",
                    Cost = 2
                },
                new ProductWithoutId
                {
                    Name = "Glasses",
                    Cost = 12
                }
            };
            return products;
        }
    }
}
